File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int answer = 0;
    int a=0,b=0;
    string s_1="";
    
    for(int i=0; i<s.size(); i++){
        
        if(s_1==""){
            s_1=to_string(s[i]);
            a++;
            continue;
        }
        
        if(s_1==to_string(s[i])) a++;
        else b++;
        
        if(a==b){
            answer++;
            a=0;b=0;
            s_1="";
        }
    }

    if(a!=0) answer+=1;
    
    return answer;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    str=''
    p,r=0,0
    for i in s :
        if str=='':
            str+=i
            p+=1
            continue
            
        if str==i :
            p+=1
        else : 
            r+=1
        
        if(p==r) :
            answer+=1
            p,r=0,0
            str=''
    
    if p!=0 : answer+=1
    return answer